gematria/datasets/exegesis-benchmark.cc

```diff
@@ -0,0 +1,256 @@
+      std::optional<int64_t> MemoryDefinitionHexValue =
+          MemoryDefinitionObject->getInteger("Value");
+
+      if (!MemoryDefinitionName.has_value() ||
+          !MemoryDefinitionSize.has_value() ||
+          !MemoryDefinitionHexValue.has_value())
+        ExitWithError("Malformed memory definition");
+
+      MemoryValue MemVal;
+      MemVal.Value = APInt(64, *MemoryDefinitionHexValue);
+      MemVal.Index = 0;  // Update this to support multiple definitions
+      MemVal.SizeBytes = *MemoryDefinitionSize;
+
+      BenchCode.Key.MemoryValues[MemoryDefinitionName->str()] = MemVal;
+    }
+
+    const json::Array *MemoryMappings =
+        TestValue.getAsObject()->getArray("MemoryMappings");
+
+    if (!MemoryMappings) ExitWithError("Malformed memory mapping");
+
+    for (const auto &MemoryMappingValue : *MemoryMappings) {
+      const json::Object *MemoryMappingObject =
+          MemoryMappingValue.getAsObject();
+
+      if (!MemoryMappingObject) ExitWithError("Malformed memory mapping");
+
+      std::optional<StringRef> MemoryMappingDefinitionName =
+          MemoryMappingObject->getString("Value");
+      std::optional<uintptr_t> MemoryMappingAddress =
+          MemoryMappingObject->getInteger("Address");
+
+      if (!MemoryMappingDefinitionName.has_value() ||
+          !MemoryMappingAddress.has_value())
+        ExitWithError("Malformed memory mapping");
+
+      MemoryMapping MemMap;
+      MemMap.Address = *MemoryMappingAddress;
+      MemMap.MemoryValueName = MemoryMappingDefinitionName->str();
+      BenchCode.Key.MemoryMappings.push_back(MemMap);
+    }
+
+    SmallVector<Benchmark, 2> AllResults;
+
+    BenchmarkRunner::RunnableConfiguration RC1 = ExitOnErr(
+        Runner->getRunnableConfiguration(BenchCode, 5000, 0, *SnipRepetitor));
+    BenchmarkRunner::RunnableConfiguration RC2 = ExitOnErr(
+        Runner->getRunnableConfiguration(BenchCode, 10000, 0, *SnipRepetitor));
+
+    std::pair<Error, Benchmark> BenchmarkResult1OrErr =
+        Runner->runConfiguration(std::move(RC1), {});
+
+    if (std::get<0>(BenchmarkResult1OrErr)) {
+      ExitOnErr(std::move(std::get<0>(BenchmarkResult1OrErr)));
+    }
+
+    AllResults.push_back(std::move(std::get<1>(BenchmarkResult1OrErr)));
+
+    std::pair<Error, Benchmark> BenchmarkResult2OrErr =
+        Runner->runConfiguration(std::move(RC2), {});
+
+    if (std::get<0>(BenchmarkResult2OrErr))
+      ExitOnErr(std::move(std::get<0>(BenchmarkResult2OrErr)));
+
+    AllResults.push_back(std::move(std::get<1>(BenchmarkResult2OrErr)));
+
+    std::unique_ptr<ResultAggregator> ResultAgg =
+        ResultAggregator::CreateAggregator(
+            Benchmark::RepetitionModeE::MiddleHalfLoop);
+
+    Benchmark Result = std::move(AllResults[0]);
+
+    ResultAgg->AggregateResults(Result,
+                                ArrayRef<Benchmark>(AllResults).drop_front());
+
+    dbgs() << Result.Measurements[0].PerSnippetValue << "\n";
+
+    dbgs() << *HexValue << "\n";
```

Comment:
The outputs (the throughput numbers) should go either to a file, or to stdout (to make it easy to filter them out of any possible debug messages that go to stderr).

Reply:
I've printed them to `STDOUT` now in the BHive format. Eventually we will probably want some explicit output file, but for now I think this is fine.